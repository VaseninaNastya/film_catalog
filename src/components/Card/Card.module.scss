.filmListCard_imgWrapper {
  display: flex;
  align-items: center;
  justify-content: center;
}
.filmListCard_img{
    max-width: 100%;
    max-height: 100%;
}
.filmListCard {
  display: flex;
  flex-direction: column;
  margin: 5px;
  position: relative;
  height: calc(100% - 10px);

}
.filmListCard:hover {
  cursor: pointer;
  box-shadow: 0 0 10px rgba(0,0,0,0.2);
}

.filmListCard_list {
  width: 70%;
  margin-left: 10px;
}



.filmListCard_title {
  height: 50px;
  overflow: hidden;
  text-align: center;
  line-clamp: 2;
  font-weight: 700;
}
.filmListCard_header{
  display: flex;
  flex-direction: column;
  flex-grow: 1;
  margin: 0 5px;
  margin-top: 5px;
}