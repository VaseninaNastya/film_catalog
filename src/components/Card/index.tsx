import React from "react";
import s from "./Card.module.scss";
import { Link } from "react-router-dom";
import { useDispatch, useSelector } from "react-redux";
import { type } from "os";
import {requestOneFilmAction} from '../../actions/getOneFilmDataAction'
interface  CardProps {
  id: string,
  title: string,
  year: string,
  type:string,
  poster:string,
}
const Card: React.FC<CardProps> = ({id, title, year, type, poster}) => {
  const dispatch = useDispatch();
  const handleGetFilm = ({currentTarget}: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
    const result = currentTarget.getAttribute("id")
    dispatch(requestOneFilmAction(result))
  }
  return(
    <Link to="/film">
      <div id = {id}  onClick = {handleGetFilm} className={s.filmListCard}>
        <ul className = {s.filmListCard_header}>
          <li key="1"><h3>{title}</h3></li>
          <li key="2"><span>year: {year}</span></li>
          <li key="3"><span>type: {type}</span></li>
        </ul>
        <div className={s.filmListCard_imgWrapper}>
          <img className={s.filmListCard_img} src={poster} alt={title}/>
        </div>
      </div>
    </Link>
  )
}

export default Card;
