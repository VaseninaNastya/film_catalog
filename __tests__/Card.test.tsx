import React from 'react';
import renderer from 'react-test-renderer'
import Card from '../src/components/Card'
import configureStore from 'redux-mock-store'
import { MemoryRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import Enzyme, { render} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';




describe('Card',()=>{
  it('renders correctly', () => {
    const mockingData = JSON.parse(`{"id" : "tt0371746", "title" : "Iron Man", "year" : "2008", "type" : "movie", "poster" : "https://m.media-amazon.com/images/M/MV5BMTczNTI2ODUwOF5BMl5BanBnXkFtZTcwMTU0NTIzMw@@._V1_SX300.jpg"}`)
  const mockStore = configureStore()
  const store = mockStore()
  Enzyme.configure({ adapter: new Adapter() })
    const tree = render(
        <Provider store={store}>
          <Router>
            <Card 
              id={mockingData.id}
              title={mockingData.title}
              year={mockingData.year}
              type={mockingData.type}
              poster = {mockingData.poster}
            />
          </Router>
        </Provider>
      )
    expect(tree).toMatchSnapshot();
  });
});
