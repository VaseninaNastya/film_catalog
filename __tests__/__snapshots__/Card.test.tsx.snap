// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
<a
  href="/film"
>
  <div
    class="filmListCard"
    id="tt0371746"
  >
    <ul
      class="filmListCard_header"
    >
      <li>
        <h3>
          Iron Man
        </h3>
      </li>
      <li>
        <span>
          year: 2008
        </span>
      </li>
      <li>
        <span>
          type: movie
        </span>
      </li>
    </ul>
    <div
      class="filmListCard_imgWrapper"
    >
      <img
        alt="Iron Man"
        class="filmListCard_img"
        src="https://m.media-amazon.com/images/M/MV5BMTczNTI2ODUwOF5BMl5BanBnXkFtZTcwMTU0NTIzMw@@._V1_SX300.jpg"
      />
    </div>
  </div>
</a>
`;
